TombEngine/Objects/TR3/Entity/tr3_civvy.cpp

```diff
@@ -2,418 +2,594 @@
 #include "Objects/TR3/Entity/tr3_civvy.h"
 
 #include "Game/control/box.h"
+#include "Game/control/lot.h"
 #include "Game/effects/effects.h"
-#include "Game/Lara/lara.h"
-#include "Game/itemdata/creature_info.h"
 #include "Game/items.h"
+#include "Game/itemdata/creature_info.h"
+#include "Game/Lara/lara.h"
 #include "Game/misc.h"
 #include "Game/people.h"
+#include "Math/Random.h"
 #include "Sound/sound.h"
 #include "Math/Math.h"
+
 #include "Specific/level.h"
 #include "Specific/setup.h"
 
 using namespace TEN::Math;
 
 namespace TEN::Entities::Creatures::TR3
 {
-	constexpr auto CIVVY_ATTACK_DAMAGE = 40;
-	constexpr auto CIVVY_SWIPE_DAMAGE  = 50;
+	constexpr auto CIVVY_ATTACK_DAMAGE  = 50;
 
-	constexpr auto CIVVY_ATTACK0_RANGE = SQUARE(BLOCK(3));
-	constexpr auto CIVVY_ATTACK1_RANGE = SQUARE(BLOCK(0.67f));
-	constexpr auto CIVVY_ATTACK2_RANGE = SQUARE(BLOCK(1));
-	constexpr auto CIVVY_WALK_RANGE	   = SQUARE(BLOCK(1));
-	constexpr auto CIVVY_ESCAPE_RANGE  = SQUARE(BLOCK(3));
-	constexpr auto CIVVY_AWARE_RANGE   = SQUARE(BLOCK(1));
+	constexpr auto CIVVY_WALK_TURN_RATE_MAX = ANGLE(5.0f);
+	constexpr auto CIVVY_RUN_TURN_RATE_MAX = ANGLE(6.0f);
+	constexpr auto CIVVY_AIM_TURN_RATE_MAX = ANGLE(8.0f);
+
+	constexpr auto CIVVY_ATTACK_WALKING_PUNCH_RANGE = SQUARE(BLOCK(0.8f));
+	constexpr auto CIVVY_ATTACK_CLOSE_PUNCH_RANGE = SQUARE(BLOCK(0.5f));
+	constexpr auto CIVVY_ATTACK_FAR_PUNCH_RANGE = SQUARE(BLOCK(0.7f));
+	
+	constexpr auto CIVVY_WALK_RANGE	   = SQUARE(BLOCK(2));
+	constexpr auto CIVVY_ESCAPE_RANGE  = SQUARE(BLOCK(5));
+	constexpr auto CIVVY_AWARE_RANGE   = SQUARE(BLOCK(15));
 
 	constexpr auto CIVVY_WAIT_CHANCE	   = 1.0f / 128;
-	constexpr auto CIVVY_STATE_WALK_CHANCE = 1.0f / 128; // Unused.
 
 	constexpr auto CIVVY_VAULT_SHIFT = 260;
 
-	constexpr auto CIVVY_WALK_TURN_RATE_MAX = ANGLE(5.0f);
-	constexpr auto CIVVY_RUN_TURN_RATE_MAX	= ANGLE(6.0f);
+	constexpr auto CIVVY_TARGET_ALERT_VELOCITY = 10.0f;
 
-	const auto CivvyBite = BiteInfo(Vector3::Zero, 13);
-	const auto CivvyAttackJoints = std::vector<unsigned int>{ 10, 13 };
+	const auto CivvyBiteRight = BiteInfo(Vector3::Zero, 13);
+	const auto CivvyBiteLeft = BiteInfo(Vector3::Zero, 10);
+	const std::vector<unsigned int> CivvyAttackJoints = { 10, 13 };
```

Comment:
Since TEN conventions prefer using `auto` wherever possible, we can easily do this: `const auto CivvyAttackJoints = std::vector<unsigned int>{ 10, 13 };`

Joints 10 and 13 could also be derived from bite structs: `{ CivvyBiteLeft.JointIndex, CivvyBiteRight.JointIndex }` (unsure about the index member variable names but something like this).

Reply:
So I got something like this:


```c++
struct CivvyBiteStruct
{
	unsigned int RightPunch = 13;
	unsigned int LeftPunch = 10;
};
	
CivvyBiteStruct CivvyBitesMeshes;

const auto CivvyBiteRight = BiteInfo(Vector3::Zero, CivvyBitesMeshes.RightPunch);
const auto CivvyBiteLeft = BiteInfo(Vector3::Zero, CivvyBitesMeshes.LeftPunch);
const auto CivvyAttackJoints = std::vector<unsigned int>{ CivvyBitesMeshes.LeftPunch, CivvyBitesMeshes.RightPunch };
```
Should make easier to change the meshes if one day were needed.